frontend/packages/console-shared/src/utils/application-utils.ts

```diff
@@ -0,0 +1,77 @@
+import { k8sList, k8sPatch, K8sResourceKind, modelFor } from '@console/internal/module/k8s';
+import * as _ from 'lodash';
+
+const updateItemAppLabel = (item: K8sResourceKind, application) => {
+  const labels = _.cloneDeep(item.metadata.labels || {});
+  const model = modelFor(item.kind);
+
+  if (!model) {
+    return Promise.reject();
+  }
+
+  if (application) {
+    labels['app.kubernetes.io/part-of'] = application;
+  } else {
+    labels['app.kubernetes.io/part-of'] = undefined;
+  }
+
+  const patch = [
+    {
+      op: _.isEmpty(labels) ? 'add' : 'replace',
+      path: '/metadata/labels',
+      value: labels,
+    },
+  ];
+
+  return k8sPatch(model, item, patch);
+};
+
+export const updateResourceApplication = (
+  resource: K8sResourceKind,
+  application: string,
+): Promise<any> => {
+  if (!resource) {
+    return Promise.reject();
+  }
+
+  const instanceName = _.get(resource, ['metadata', 'labels', 'app.kubernetes.io/instance']);
+  const prevApplication = _.get(resource, ['metadata', 'labels', 'app.kubernetes.io/part-of']);
+
+  const lists: Promise<any>[] = [];
+  const patches: Promise<any>[] = [updateItemAppLabel(resource, application)];
+
+  // If there is no instance label, only update this item
+  if (!instanceName) {
+    return Promise.all(patches);
+  }
+
+  // Retrieve all resources with the same instance label
+  const kinds = ['ReplicationController', 'Route', 'Service', 'ReplicaSet', 'BuildConfig', 'Build'];
```

Comment:
Do we need to list kinds? What will happen with knative resources?

Reply:
Each `kind` needs to be retrieved separately. From what I can tell, there are no knative resources that use the 'app.kubernetes.io/part-of' label.

Reply:
@invincibleJai do serverless resources adhere to the labeling conventions? You know the labels we apply at import time. And if so, does the controller propagate those labels to the child resources it creates?